app/browser/dataset-browse.tsx

```diff
@@ -622,29 +621,23 @@ export const SearchFilters = ({ data }: { data?: DataCubesQuery }) => {
     },
   });
 
-  const allCounts = useDatasetCount(filters, includeDrafts);
-  const resultsCounts = useMemo(() => {
+  const counts = useMemo(() => {
     if (!data?.dataCubes) {
       return {};
-    } else {
-      const res = {} as Record<string, number>;
-      for (const cube of data.dataCubes) {
-        const countables = [
-          ...cube.dataCube.themes,
-          cube.dataCube.creator,
-        ].filter(truthy);
-        for (const item of countables) {
-          res[item.iri] = res[item.iri] || 0;
-          res[item.iri] += 1;
-        }
+    }
+
+    const result: Record<string, number> = {};
+
+    for (const { dataCube } of data.dataCubes) {
+      const countables = [...dataCube.themes, dataCube.creator].filter(truthy);
+
+      for (const { iri } of countables) {
+        result[iri] = (result[iri] ?? 0) + 1;
```

Comment:
I think the idea behind the count was to not have to retrieve all the matching data cubes in case there was a really long list of data cubes. Isn't it a problem now ?

Reply:
What's weird to me is that counting if done via SPARLQ request should not be done via another query from the front-end, and should be done on the server as part of the search query. Don't know why I did it this way before 🤷 .

Reply:
I think that could become a problem if we implement pagination, and put a limit on the number of cubes we fetch at once; then we would need to have a way to retrieve the total counts separately 👍 

However now that the search query takes ~1.5s to fetch 263 cubes (INT + drafts), and we only have 64 on PROD, we could even do the pagination purely on the client side, and move to server side once it becomes a performance problem. But we didn't discuss pagination at any point yet, and in the current form this query was kinda an "extra one" 😄